### PR TITLE
adapted the changes made from the other branch to this one, and it is…

### PR DESCRIPTION
**Pull request by VladLoghin**
… all done, the visit start date and end date have been implemented in the system.

Fixed some issues with the update page

fixed some issues had with the get by id in the front-end

[**JIRA:** link to jira ticket
](https://champlainsaintlambert.atlassian.net/browse/CPC-807?atlOrigin=eyJpIjoiNmE0MjkyMTM3NWY3NDlkOWI2NjJjMTMzMWEwNzBiYzAiLCJwIjoiaiJ9)
## Context:
The system had not yet implemented a way to determine when an appointment was going to end. With this change it now does
## Does this PR change the .vscode folder in petclinic-frontend?:
the PR will also add the visit end date to the list of fields when making requests, showing that the visit ends in an hour.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes

- This changes the UI a bit providing more information for the Admin to process.
- The response model now accepts an extra field.

## Before and After UI (Required for UI-impacting PRs)
image

image

## Dev notes (Optional)
Specific technical changes that should be noted
## Linked pull requests (Optional)
Pull request links

### CODE REVIEW

**Review by AlejandroBrnab — APPROVED** [review]
Thank you for making another branch to do the PR and thank you for taking the time to put your files on it. Your other PR had a lot of files changed and other stuff that could have possibly messed up main.

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/DataLayer/DataSetupService.java
Comment (FelixAllard) [inline visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/DataLayer/DataSetupService.java:1]: Modified the database which is really good!

**Review by FelixAllard — APPROVED** [review]
I see the field was added everywhere. StartVisitDate was also added to the database, which is good. No problem with the CI testing. Looks good to me! Just be careful about changing the package.json versions...

File: petclinic-frontend/package.json [hunk 1]
```diff
@@ -25,9 +25,9 @@
   "devDependencies": {
     "@originjs/vite-plugin-federation": "1.3.5",
     "@types/node": "20.12.12",
-    "@types/react": "18.3.3",
+    "@types/react": "^18.3.3",
     "@types/react-bootstrap": "^0.32.37",
-    "@types/react-dom": "18.3.0",
+    "@types/react-dom": "^18.3.0",
```
Comment (Rrregine) [inline petclinic-frontend/package.json:30]: It this a miss type mistake? Why there is ^ mark?

**Review by Rrregine — APPROVED** [review]
Basically you just change the field "visitDate" to "VisitStartDate. You also add a "visitEndDate". It is very good that you didn't touch anything else!

**Pull request merged**